app/src/main/kotlin/at/bitfire/davdroid/sync/account/AccountsCleanupWorker.kt

```diff
@@ -70,32 +69,17 @@ class AccountsCleanupWorker @AssistedInject constructor(
 
         // Later, accounts which are not in the DB should be deleted here
 
-        val mainAccountType = applicationContext.getString(R.string.account_type)
-        val mainAccountNames = accounts
-            .filter { account -> account.type == mainAccountType }
+        val accountType = applicationContext.getString(R.string.account_type)
+        val accountNames = accounts
+            .filter { account -> account.type == accountType }
             .map { it.name }
-
-        val addressBookAccountType = applicationContext.getString(R.string.account_type_address_book)
-        val addressBooks = accounts
-            .filter { account -> account.type == addressBookAccountType }
-            .map { addressBookAccount -> LocalAddressBook(applicationContext, addressBookAccount, null) }
-        for (addressBook in addressBooks) {
-            try {
-                val mainAccount = addressBook.mainAccount
-                if (mainAccount == null || !mainAccountNames.contains(mainAccount.name))
-                    // the main account for this address book doesn't exist anymore
-                    addressBook.deleteCollection()
-            } catch(e: Exception) {
-                logger.log(Level.SEVERE, "Couldn't delete address book account", e)
-            }
-        }
```

Comment:
Instead of this, we should iterate through all address book accounts and check their collection IDs. If the collection doesn't exist anymore, the address book account should also be deleted.

However this should be done _after_ the services have been deleted below.

Reply:
Yes, plausible. I forgot about that.